ui/src/views/repositories/index.tsx

```diff
@@ -14,16 +14,27 @@ import Loading from 'src/components/Loading'
 import { showErrorAlert } from 'src/utils/alerts'
 
 const RepositoriesView: React.FC = () => {
-  const [{ showAddRepositoryModal, showAutoImportModal, showSyncRepoModal }] = useRepositoriesContext()
+  const [showTable, setShowTable] = useState(false)
+
+  const [{ showAddRepositoryModal, showAutoImportModal, showSyncRepoModal, search }] = useRepositoriesContext()
 
   const { setShowAutoImportModal, setShowSyncRepoModal } = useRepositoriesSetState()
 
-  const { loading, error, data } = useQuery(GET_REPOS, {
+  // - Todo: connect selectedRepositoriesCount from RepositoriesTable
+  const selectedRepositoriesCount = 0
+
+  const { loading, error, data, refetch } = useQuery(GET_REPOS, {
+    variables: { search },
     pollInterval: 5000,
   })
 
-  // - Todo: connect selectedRepositoriesCount from RepositoriesTable
-  const selectedRepositoriesCount = 0
+  useEffect(() => {
+    refetch({ search })
+  }, [refetch, search])
+
+  if (data && !showTable) {
+    setShowTable(true)
+  }
```

Comment:
this seems a little _funky_ 😃 to me - calling a `setXxx` function in the render method - feels like an antipattern? But I'm not entirely sure. Should it be in a `useEffect` or should we derive whether to show the table in the returned  JSX?

Reply:
mmm yeah, it looks some weird but I had to left it like so. I had it inside a `useEffect` at first but its behavior was no so good because it rendered for a short time `<Empty Repository Table />`, that's because it had to wait for `useQuery` to be resolved. Not sure if it is an anti-pattern but at the end was the best way to get a good user experience. The good thing is that `setShowTable(true)` is executed just one time (at first render)

Reply:
Done :smiley: [89a363c](https://github.com/mergestat/fuse/pull/135/commits/89a363c1e1213ece491ae06301b73efa408ba1fc)